src/api/service/Screenings.js

```diff
@@ -0,0 +1,10 @@
+import axios from "@/api/axiosConfig.js";
+import { SCREENINGS } from "@/api/apiEndpoints.js";
+
+export const getAllScreenings = () => {
+  return axios.get(SCREENINGS)
+}
+
+export const getScreeningsByMovieAndDate = (movieId, date) => {
+  return axios.get(`${SCREENINGS}?movie_id=${movieId}&date=${date}`);
+};
```

Comment:
Ideally it would be better if we only have one of those here (the second one!) and treat the arguments as optional and handle that case.

Also, for the query params axios can handle them in a much better way 😄
https://masteringjs.io/tutorials/axios/get-query-params

Reply:
Do you mean something like this?

`export const getScreeningsByMovieAndDate = (movieId, screeningsDate) => {
  const params = {
    movie_id: movieId,
    date: screeningsDate
  }

  if (movieId === undefined) {
    return axios.get(`${SCREENINGS}?date=${screeningsDate}`)
  }
  return axios.get(SCREENINGS, { params });
};`

Reply:
I think you could actually safely just do
```js
const params = {
  movie_id: movieId,
  date: screeningsDate
}

return axios.get(SCREENINGS, { params });
```

If the params are undefined - axios won't add them to the query. Nice and easy 😄

Reply:
Magic! 😮 🧙‍♂️ 😄